### ISSUE TITLE
🐛 BUG: `<style>` dependencies are not tracked

### ISSUE THREAD

**Issue by bradlc** [snowpackjs/astro#480]
### Bug Report: Quick Checklist

- [x] I am on the latest version of Astro & all plugins.
- [x] I am on an ESM Node.js version (`v12.20.0+, v14.13.1+, or v16.0.0+`)
- [x] I use package manager: npm 7.7.6
- [x] I run Astro on OS: macOS 11.2.3

### Describe the bug

`<style>` blocks in Astro components can have dependencies. For example you might use a SCSS import:

```html
<!-- src/pages/index.astro -->
<style lang="scss">
@import '../styles/example.scss';
</style>
```

Here, `src/pages/index.astro` depends on `src/styles/example.scss`. If `example.scss` changes then `index.astro` needs to be recompiled. Currently dependencies are not tracked so changes to them do not invalidate the component compilation. 

### Steps to reproduce

1. Clone https://github.com/bradlc/astro-style-dependencies
2. Run `npm install`
3. Run `npm start`
4. Open `localhost:3000` and notice that the box is hotpink
5. Edit `src/styles/example.scss`, changing `hotpink` to `tomato`
6. Refresh the browser window and notice that the box is still hotpink, when it should now be tomato

### Expected behavior

Changes to `example.scss` should be reflected without having to re-save the Astro component.

### Anything else?

The example above and the reproduction repository use SCSS imports as an example but [PostCSS plugins can declare dependencies](https://github.com/postcss/postcss/blob/main/docs/guidelines/runner.md#3-dependencies) as well, and these should also be tracked.

**Comment by drwpow** [comment]
Thanks for raising. I’m seeing this behavior. Will look into a fix!

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
